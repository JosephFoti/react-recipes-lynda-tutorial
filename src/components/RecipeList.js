import React from 'react';
import PropTypes from 'prop-types';
import RecipeListItem from './RecipeListItem';

const RecipeList = ({ style, favorites, recipes, ...props }) => (
  <ul className="list-reset">
    {recipes.map(recipe => (
      <RecipeListItem
        key={recipe.id}
        recipe={recipe}
        favorited={favorites.includes(recipe.id)}
        {...props}
      />
    ))}
  </ul>
);

RecipeList.propTypes = {
  recipes: PropTypes.array,
  style: PropTypes.object,
  favorites: PropTypes.array,
};

RecipeList.defaultProps = {
  recipes: [],
  favorites: [],
};

export default RecipeList;
