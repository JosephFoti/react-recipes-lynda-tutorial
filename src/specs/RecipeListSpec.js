import React from 'react';
import renderer from 'react-test-renderer';
import RecipeList from '../components/RecipeList';

const testRecipes = [
  {
    id: 1,
    name: 'Test Recipe',
    category: 'Test category',
  },
  {
    id: 2,
    name: 'Test Recipe 2',
    category: 'Test category 2',
  },
];

describe('<RecipeList />', () => {
  test('Should not break withour recipe', () => {
    const component = renderer.create(<RecipeList />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Render correctly', () => {
    const component = renderer.create(<RecipeList recipes={testRecipes} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Render favorite correctly', () => {
    const component = renderer.create(
      <RecipeList recipes={testRecipes} favorites={[1, 2, 3]} />,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
