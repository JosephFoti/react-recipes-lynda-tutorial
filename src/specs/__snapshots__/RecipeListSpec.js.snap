// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeList /> Render correctly 1`] = `
<ul
  className="list-reset"
>
  <li
    className="py2 border-bottom border-bottom-dashed border-width pointer"
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      ◻️
    </span>
    <span>
      Test Recipe
    </span>
    <span>
      Test category
    </span>
  </li>
  <li
    className="py2 border-bottom border-bottom-dashed border-width pointer"
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      ◻️
    </span>
    <span>
      Test Recipe 2
    </span>
    <span>
      Test category 2
    </span>
  </li>
</ul>
`;

exports[`<RecipeList /> Render favorite correctly 1`] = `
<ul
  className="list-reset"
>
  <li
    className="py2 border-bottom border-bottom-dashed border-width pointer"
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      ✅
    </span>
    <span>
      Test Recipe
    </span>
    <span>
      Test category
    </span>
  </li>
  <li
    className="py2 border-bottom border-bottom-dashed border-width pointer"
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      ✅
    </span>
    <span>
      Test Recipe 2
    </span>
    <span>
      Test category 2
    </span>
  </li>
</ul>
`;

exports[`<RecipeList /> Should not break withour recipe 1`] = `
<ul
  className="list-reset"
/>
`;
